src/query/service/src/interpreters/interpreter_table_modify_column.rs

```diff
@@ -313,158 +313,149 @@ impl ModifyTableColumnInterpreter {
-                                box TableDataType::Tuple {
-                                    fields_name: new_fields_name,
-                                    fields_type: new_fields_type,
-                                },
-                            )),
-                        ) => {
-                            let transform_funcs = new_fields_name
-                                .iter()
-                                .zip(new_fields_type.iter())
-                                .map(|(new_field_name, new_field_type)| {
-                                    match old_fields_name.iter().position(|n| n == new_field_name) {
-                                        Some(idx) => {
-                                            format!(
-                                                "array_transform(`{}`, v -> v.{})",
-                                                field.name,
-                                                idx + 1
-                                            )
-                                        }
-                                        None => {
-                                            let new_data_type = DataType::from(new_field_type);
-                                            let default_value =
-                                                Scalar::default_value(&new_data_type);
-                                            format!("{default_value}")
-                                        }
-                                    }
-                                })
-                                .collect::<Vec<_>>()
-                                .join(", ");
-
-                            format!(
-                                "if(is_not_null(`{}`), arrays_zip({}), NULL) AS {}",
-                                field.name, transform_funcs, field.name
-                            )
-                        }
-                        (_, _) => {
-                            if need_remove_nullable {
-                                // If the column is being changed from NULLABLE to NOT NULL,
-                                // wrap it with `coalesce()` to replace NULL values with the default,
-                                // and `remove_nullable()` to mark the resulting expression as non-nullable.
-                                format!(
-                                    "remove_nullable(coalesce(`{}`, {}))",
-                                    field.name, default_scalar
-                                )
-                            } else {
-                                format!("`{}`", field.name)
+        let mut query_fields = Vec::new();
+        for field in new_schema_without_computed_fields.fields() {
+            let old_field = schema.field_with_name(&field.name)?;
+
+            // Check for NULL values in columns that are being changed to NOT NULL
+            if old_field.data_type.is_nullable() && !field.data_type.is_nullable() {
```

Comment:
There's no need to write this check in a loop, because we only change the type of one column